libs/network/esp8266/jswrap_esp8266.c

```diff
@@ -1897,29 +1939,38 @@ static void scanCB(void *arg, STATUS status) {
     // sint16 freq_offset
     // ---
     // Create, populate and add a child ...
-    JsVar *currentAccessPoint = jspNewObject(NULL, "AccessPoint");
-    jsvObjectSetChildAndUnLock(currentAccessPoint, "rssi", jsvNewFromInteger(bssInfo->rssi));
-    jsvObjectSetChildAndUnLock(currentAccessPoint, "channel", jsvNewFromInteger(bssInfo->channel));
-    jsvObjectSetChildAndUnLock(currentAccessPoint, "authMode", jsvNewFromInteger(bssInfo->authmode));
-    jsvObjectSetChildAndUnLock(currentAccessPoint, "isHidden", jsvNewFromBool(bssInfo->is_hidden));
+    JsVar *jsCurrentAccessPoint = jspNewObject(NULL, "AccessPoint");
+    jsvObjectSetChildAndUnLock(jsCurrentAccessPoint, "rssi", jsvNewFromInteger(bssInfo->rssi));
+    jsvObjectSetChildAndUnLock(jsCurrentAccessPoint, "channel", jsvNewFromInteger(bssInfo->channel));
+    jsvObjectSetChildAndUnLock(jsCurrentAccessPoint, "authMode", jsvNewFromInteger(bssInfo->authmode));
+    jsvObjectSetChildAndUnLock(jsCurrentAccessPoint, "isHidden", jsvNewFromBool(bssInfo->is_hidden));
     // The SSID may **NOT** be NULL terminated ... so handle that.
     char ssid[sizeof(bssInfo->ssid) + 1];
     os_strncpy((char *)ssid, (char *)bssInfo->ssid, sizeof(bssInfo->ssid));
     ssid[sizeof(ssid)-1] = '\0';
-    jsvObjectSetChildAndUnLock(currentAccessPoint, "ssid", jsvNewFromString(ssid));
+    jsvObjectSetChildAndUnLock(jsCurrentAccessPoint, "ssid", jsvNewFromString(ssid));
 
     // Add the new record to the array
-    jsvArrayPush(accessPointArray, currentAccessPoint);
+    jsvArrayPush(jsAccessPointArray, jsCurrentAccessPoint);
+    jsvUnLock(jsCurrentAccessPoint);
 
     os_printf(" - ssid: %s\n", bssInfo->ssid);
     bssInfo = STAILQ_NEXT(bssInfo, next);
   }
 
   // We have now completed the scan callback, so now we can invoke the JS callback.
-  JsVar *params[1];
-  params[0] = accessPointArray;
-  jsiQueueEvents(NULL, g_jsScanCallback, params, 1);
+  // The parameters to the callback are:
+  // * err - An error indication - always null as we can't fail.
+  // * accessPointArray - An array of access point records.
+  JsVar *params[2];
+  params[0] = jsvNewNull();
+  params[1] = jsAccessPointArray;
+  jsiQueueEvents(NULL, g_jsScanCallback, params, 2);
+
+  jsvUnLock(jsAccessPointArray);
```

Comment:
Minor thing here, you can always use `jsvUnlock3` to unlock a bunch at once. It's not such an issue on ESP8266, but it actually saves a bunch of code space when it's done on other platforms.